irene/Percolation.java

```diff
@@ -0,0 +1,110 @@
+import edu.princeton.cs.algs4.WeightedQuickUnionUF;
+
+/*---------------------------------------------------------------------------------------------------------
+ *  By Irene Storozhko. August 11, 2018. Week 1 assignment for Algorithms I course by Wayne and Sedgewick.
+ *
+ *  Percolation data type to model a percolation system.
+ *  IMPORTANT: Coordinates: arow, acol belong to [0; n - 1] - these are for the internal use;
+ *                          row, col belong to [1, n] - these for the exposed API.
+ *--------------------------------------------------------------------------------------------------------*/
+public class Percolation {
+    private boolean[][] grid; // Models the grid
+    private final WeightedQuickUnionUF percolationTree; // Models the percolation of the system
+    private final WeightedQuickUnionUF fullnessTree;
+    private final int virtualTop; // Supposedly, helps quickly determine whether the system percolates
+    private final int virtualBottom; // Supposedly, helps quickly determine whether the system percolates
+    private int numberOfOpen; // Number of open sites in the system
```

Comment:
I would use the same name for the method and for the variable. In this case you don't need to put a comment here because the variable name would explain itself